docs/source/python/api/compute.rst

```diff
@@ -48,24 +48,25 @@ Aggregations
 Grouped Aggregations
 --------------------
 
-.. autosummary::
-   :toctree: ../generated/
-
-   hash_all
-   hash_any
-   hash_approximate_median
-   hash_count
-   hash_count_distinct
-   hash_distinct
-   hash_max
-   hash_mean
-   hash_min
-   hash_min_max
-   hash_product
-   hash_stddev
-   hash_sum
-   hash_tdigest
-   hash_variance
+The grouped aggregation functions are not directly invokable, but are used
+as part of a SQL-style "group by" operation using the
+:meth:`~pyarrow.Table.group_by` method. A list of the available aggregations:
+
+* ``hash_all``
+* ``hash_any``
+* ``hash_approximate_median``
+* ``hash_count``
+* ``hash_count_distinct``
+* ``hash_distinct``
+* ``hash_max``
+* ``hash_mean``
+* ``hash_min``
+* ``hash_min_max``
+* ``hash_product``
+* ``hash_stddev``
+* ``hash_sum``
+* ``hash_tdigest``
+* ``hash_variance``
```

Comment:
I think we will have to also add the docstring content somehow. Otherwise we will be losing the column where we explain what the function does in the output.

![sshot](https://user-images.githubusercontent.com/601423/144229330-5d9929fa-bd64-4492-ac36-b1e40242ee9b.png)

Reply:
(IMO then we could also simply not hide the functions, and keep them available for introspection like that)

Reply:
FYI, I think we can now remove them from the API Reference (in the end if you can't invoke them it's not true that they are an available api) and I ended up adding the list of them in the compute documentation improvement I was working on: https://github.com/apache/arrow/pull/11830/files

Reply:
Maybe you can remove this list in https://github.com/apache/arrow/pull/11830 (or replace it with your custom directive), instead of merging this PR?

Reply:
Sounds good to me, feel free to close this one.